feat: set cmdheight=0 (#16251)
neovim/neovim: runtime/doc/options.txt

	The value of this option is stored with the tab page, so that each tab
	page can have a different value.

> 	When 'cmdheight' is zero, it disables echo area and all outputs needs

Reviewer: ```suggestion
	When 'cmdheight' is 0, it disables echo area and all outputs needs
```

I think we should use `0` here instead of `zero` for consistency with other places.

	The value of this option is stored with the tab page, so that each tab
	page can have a different value.

	When 'cmdheight' is zero, it disables echo area and all outputs needs

> 	|hit-enter| prompts.

Reviewer 2: That doesn't seem desirable. Can be improved later I guess.

Author: It is improved by https://github.com/neovim/neovim/pull/16480.

	The value of this option is stored with the tab page, so that each tab
	page can have a different value.

> 	When 'cmdheight' is zero, it disables echo area and all outputs needs
> 	|hit-enter| prompts.

Reviewer 3: ```suggestion
	When 'cmdheight' is zero, it disables echo area and all outputs need
	|hit-enter| prompt.
```

		45%	relative position in the file
	If 'rulerformat' is set, it will determine the contents of the ruler.
	Each window has its own ruler.  If a window has a status line, the

> 	ruler is shown there.  If a window has not a status line and

Reviewer: ```suggestion
	ruler is shown there.  If a window does not have a status line and
```

		45%	relative position in the file
	If 'rulerformat' is set, it will determine the contents of the ruler.
	Each window has its own ruler.  If a window has a status line, the
	ruler is shown there.  If a window doesn't have a status line and

> 	'cmdheight' is 0, the ruler is not shown.  Otherwise it is shown in

Reviewer 3: ```suggestion
	'cmdheight' is 0, the ruler is not shown. Otherwise it is shown in
```

Reviewer 4: Using two spaces seems more consistent here.

Reviewer 3: Oh? Ignore this then

> 	the last line of the screen.  If the statusline is given by

Reviewer 3: It appears that using two spaces is more consistent, so ignore this suggestion

		45%	relative position in the file
	If 'rulerformat' is set, it will determine the contents of the ruler.
	Each window has its own ruler.  If a window has a status line, the

> 	ruler is shown there.  If a window has not a status line and

Reviewer 3: ```suggestion
	ruler is shown there.  If a window doesn't have a status line and
```

			global
	If in Insert, Replace or Visual mode put a message on the last line.
	The |hl-ModeMsg| highlight group determines the highlighting.

> 	The option is disabled if 'cmdheight' is 0.

Reviewer 2: is this tested explicitly? something like:

```
set cmdheight=0
eq('showmode', 0)
set showmode  // error? or what?
```

Author: I have added the tests.
